merge_checks/commit_checks.py

```diff
@@ -53,13 +51,13 @@ def has_wrong_commit_message(subject_markers: Sequence[str]) -> Sequence[str]:
     )
 
 
-def main(head_hash: str, base_ref: str) -> int:
+def get_commit_checks_result(head_hash: str, base_ref: str) -> tuple[bool, str]:
     fetch_head_only(base_ref)
     base_hash = get_base_revision(base_ref)
 
     if head_hash == base_hash:
         print(f"HEAD identical with {base_ref}, no commits to check")
-        return 0
+        return True, "No commits to check"
```

Comment:
Maybe then remove all prints are add one outside?

One could also argue we should migrate to logging - I started with prints, because the script was just a few lines... not anymore.

Reply:
Can you please explain more what you would like to see here, @zyv or @marns93? At the moment, I do not have enough information to start making changes.

Reply:
So at the moment in this function you have `print`-based logging. After your changes you introduced the returns that supply a success flag and corresponding message, which basically duplicate the `print` statements.

One could argue that you can then just stick the `print`-statement outside the function outputting the corresponding message and get rid of the internal `print`s, because the exact message is not important, they we there just to be able to roughly trace the code.

Further one can argue that the `print`-based logging is evil anyways, and since this script has grown a lot one should replace remaining `print` statements with logging calls.

Reply:
Thanks, @zyv. The logger was added in 515cfe98ef12c9a4b975e9e526803b23f4269be0. I'll still look into the config a bit more. Let me know if you have any perferences that you would like to see, @marns93.

Reply:
@felix11h I think your configuration is enough.